### PR TITLE
Decentralize `.gitattributes`, moving `locales.rs.data` and `segmenter_rules` out of generated directories

### PR DESCRIPTION
**Pull request by robertbastian**
`segmenter_rules` are handwritten and basically code, so I've moved them from `data` (which only contains generated data) into `src/transform/segmenter`.

### CODE REVIEW

**Review by Manishearth — APPROVED** [review]
I feel like the segmenter files have moved a bunch of times before so I'd also want shane's approval here

**Review by sffc — APPROVED** [review]
Willing to take bets for how many more times we move around these types of data files in the repo. I do agree that this seems like an appropriate place for these four toml files.

**Pull request merged**
